supabase/database/sql/get_snippets_function.sql

```diff
@@ -214,7 +209,68 @@ BEGIN
                 )
             )
         )
-        ORDER BY s.recorded_at DESC;
+        AND (
+            p_filter IS NULL OR
+            NOT p_filter ? 'upvotedBy' OR
+            (
+                CASE
+                    WHEN jsonb_array_length(p_filter->'upvotedBy') = 0 THEN TRUE
+                    ELSE (
+                        CASE
+                            WHEN (
+                                p_filter->'upvotedBy' ? 'by_me' AND
+                                p_filter->'upvotedBy' ? 'by_others'
+                            ) THEN
+                                EXISTS (
+                                    SELECT 1
+                                    FROM label_upvotes lu
+                                    WHERE lu.snippet_label IN (
+                                        SELECT id FROM snippet_labels WHERE snippet = s.id
+                                    )
+                                )
+                            WHEN p_filter->'upvotedBy' ? 'by_me' THEN
+                                EXISTS (
+                                    SELECT 1
+                                    FROM label_upvotes lu
+                                    WHERE lu.snippet_label IN (
+                                        SELECT id FROM snippet_labels WHERE snippet = s.id
+                                    )
+                                    AND lu.upvoted_by = current_user_id
+                                )
+                            WHEN p_filter->'upvotedBy' ? 'by_others' THEN
+                                EXISTS (
+                                    SELECT 1
+                                    FROM label_upvotes lu
+                                    WHERE lu.snippet_label IN (
+                                        SELECT id FROM snippet_labels WHERE snippet = s.id
+                                    )
+                                    AND lu.upvoted_by != current_user_id
+                                )
+                            ELSE FALSE
+                        END
+                    )
+                END
+            )
+        )
+        ORDER BY
+          CASE 
+          WHEN p_order_by = 'upvotes' THEN (s.upvote_count + s.like_count)
```

Comment:
Let's add some indentation here to make the `case-when` statement here more readable